src/fsw/FCCode/AttitudeController.cpp

```diff
@@ -0,0 +1,208 @@
+    pointer_vec1_current_f.set(lin::nans<lin::Vector3f>());
+    pointer_vec1_desired_f.set(lin::nans<lin::Vector3f>());
+    pointer_vec2_current_f.set(lin::nans<lin::Vector3f>());
+    pointer_vec2_desired_f.set(lin::nans<lin::Vector3f>());
+}
+
+void AttitudeController::default_all() {
+    default_actuator_commands();
+    default_pointing_objectives();
+}
+
+void AttitudeController::calculate_detumble_controller() {
+    m_body_cmd_f.set(lin::zeros<lin::Vector3f>());
+
+    // Default all inputs to NaNs and set appropriate fields
+    detumbler_data = gnc::DetumbleControllerData();
+    detumbler_data.b_body = b_body_rd_fp->get();
+
+    // Call the controller and write results to appropriate state fields
+    control_detumble(detumbler_state, detumbler_data, detumbler_actuation);
+    if (lin::all(lin::isfinite(detumbler_actuation.mtr_body_cmd)))
+        m_body_cmd_f.set(detumbler_actuation.mtr_body_cmd);
+}
+
+void AttitudeController::calculate_pointing_objectives() {
+    default_pointing_objectives();
+
+    lin::Vector3f dr_body = lin::nans<lin::Vector3f>();
+    lin::Matrix3x3f DCM_hill_body = lin::nans<lin::Matrix3x3f>();
+    {
+        lin::Vector3f r = pos_ecef_fp->get(); // r = r_ecef
+        lin::Vector3f v = vel_ecef_fp->get(); // v = v_ecef
+
+        // Position and velocity must be finite
+        if (lin::any(!(lin::isfinite(r) && lin::isfinite(v)))) return;
+
+        // Current time since the PAN epoch in seconds
+        double time = static_cast<double>(time_ns_fp->get()) * 1.0e-9;
+
+        lin::Vector4f q_body_ecef;
+        gnc::env::earth_attitude(time, q_body_ecef); // q_body_ecef = q_ecef_eci
+        gnc::utl::quat_conj(q_body_ecef);            // q_body_ecef = q_eci_ecef
+        gnc::utl::quat_cross_mult(q_body_eci_est_fp->get(), q_body_ecef);
+
+        lin::Vector3f w_earth_ecef;
+        gnc::env::earth_angular_rate(t, w_earth_ecef_eci);  // rate of ecef frame in eci
+        v = v - lin::cross(w_earth_ecef, r);                // v_ecef but intertial
+
+        gnc::utl::rotate_frame(q_body_ecef, r); // Throw the vectors into the body frame
+        gnc::utl::rotate_frame(q_body_ecef, v);
+        gnc::utl::dcm(DCM_hill_body, r, v);     // Calculate our dcm
+
+        lin::Vector3f dr = pos_baseline_ecef_fp->get(); // dr = dr_ecef
+
+        // Ensure we have a valid relative position
+        if (lin::all(lin::isfinite(dr_body))) {
+            dr_body = dr;                            // dr_body = dr_ecef
+            gnc::utl::rotate_frame(q_body_ecef, dr_body);
+        }
+    }
+
+    adcs_state_t state = static_cast<adcs_state_t>(adcs_state_fp->get());
+    switch (state) {
+        /*
+         * The general idea for standby pointing is to have the antenna face
+         * along the velocity vector (we're assuming a near circular orbt here)
+         * and have the docking face pointing normal to our orbit.
+         */
+        case adcs_state_t::point_standy:
```

Comment:
Typo here: point_standy
```suggestion
        case adcs_state_t::point_standby:
```